parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -62,6 +62,7 @@
 
   private MessageType requestedSchema;
   private MessageType fileSchema;
+  private MessageColumnIO columnIO;
```

Comment:
This is actually another problem I found during this PR although not related.
Previouslly, columnIO was not a field, and it gets re-parsed/generated for each row group. This is completely unnecessary.

I am changing it to be generated in the initialize method, and it is reused for each row group.

This change might bring back some performance when the schema is really large.

I can separate this in another PR, what do you think? @isnotinvain